platform/x11/os_x11.cpp

```diff
@@ -1529,6 +1529,15 @@ void OS_X11::set_window_size(const Size2 p_size) {
 	int old_h = xwa.height;
 
 	Size2 size = p_size;
+
+	if (Math::is_nan(size.x)) {
+		WARN_PRINT("Invalid width (NaN), using 0 instead.");
+		size.x = 0;
+	}
+	if (Math::is_nan(size.y)) {
+		WARN_PRINT("Invalid height (NaN), using 0 instead.");
+		size.y = 0;
+	}
```

Comment:
Since it's then set to 1 right after, it would be best to set it to 1 right away instead of mentioning 0.

Reply:
Good point :) I've replaced it with `ERR_FAIL_COND`, let me know if you would like to add a message as well :)